x-pack/dockerlogbeat/pipelinemanager/clientLogReader.go

```diff
@@ -49,28 +45,61 @@ func newClientFromPipeline(pipeline *pipeline.Pipeline, file, hashstring string)
 	}
 
 	// Create the FIFO reader client from the FIPO pipe
-	inputFile, err := fifo.OpenFifo(context.Background(), file, syscall.O_RDONLY, 0700)
+	inputFile, err := pipereader.NewReaderFromPath(file)
 	if err != nil {
 		return nil, errors.Wrapf(err, "error opening logger fifo: %q", file)
 	}
 	logp.Info("Created new logger for %s", file)
 
-	return &ClientLogger{logFile: inputFile, client: client, pipelineHash: hashstring}, nil
+	return &ClientLogger{logFile: inputFile, client: client, pipelineHash: hashstring, closer: make(chan struct{})}, nil
 }
 
 // Close closes the pipeline client and reader
 func (cl *ClientLogger) Close() error {
 	logp.Info("Closing ClientLogger")
-	cl.client.Close()
+	cl.logFile.Close()
+	return cl.client.Close()
 
-	return cl.logFile.Close()
 }
 
 // ConsumeAndSendLogs reads from the FIFO file and sends to the pipeline client. This will block and should be called in its own goroutine
 // TODO: Publish() can block, which is a problem. This whole thing should be two goroutines.
-func (cl *ClientLogger) ConsumeAndSendLogs() {
-	reader := pb.NewUint32DelimitedReader(cl.logFile, binary.BigEndian, 2e6)
-
+// func (cl *ClientLogger) ConsumeAndSendLogs() {
```

Comment:
?

Reply:
Ah yah, I just left in a huge chunk of commented-out code, didn't I?